src/script/miniscript.h

```diff
@@ -676,6 +744,166 @@ struct Node {
+                    for (size_t i = 0; i < sats.size(); ++i) {
+                        if (i != node.k) nsat = Choose(std::move(nsat), std::move(sats[i]), nonmal);
+                    }
+                    assert(node.k <= sats.size());
+                    return InputResult(std::move(nsat), std::move(sats[node.k]));
+                }
+                case NodeType::OLDER: {
+                    return InputResult(INVALID, ctx.CheckOlder(node.k) ? EMPTY : INVALID);
+                }
+                case NodeType::AFTER: {
+                    return InputResult(INVALID, ctx.CheckAfter(node.k) ? EMPTY : INVALID);
+                }
+                case NodeType::SHA256: {
+                    std::vector<unsigned char> preimage;
+                    Availability avail = ctx.SatSHA256(node.data, preimage);
+                    return InputResult(ZERO32, InputStack(std::move(preimage)).Available(avail));
+                }
+                case NodeType::RIPEMD160: {
+                    std::vector<unsigned char> preimage;
+                    Availability avail = ctx.SatRIPEMD160(node.data, preimage);
+                    return InputResult(ZERO32, InputStack(std::move(preimage)).Available(avail));
+                }
+                case NodeType::HASH256: {
+                    std::vector<unsigned char> preimage;
+                    Availability avail = ctx.SatHASH256(node.data, preimage);
+                    return InputResult(ZERO32, InputStack(std::move(preimage)).Available(avail));
+                }
+                case NodeType::HASH160: {
+                    std::vector<unsigned char> preimage;
+                    Availability avail = ctx.SatHASH160(node.data, preimage);
+                    return InputResult(ZERO32, InputStack(std::move(preimage)).Available(avail));
+                }
+                case NodeType::AND_V: {
+                    auto& x = subres[0], &y = subres[1];
+                    return InputResult((y.nsat + x.sat).NonCanon(), y.sat + x.sat);
+                }
+                case NodeType::AND_B: {
+                    auto& x = subres[0], &y = subres[1];
+                    return InputResult(Choose(Choose(y.nsat + x.nsat, (y.sat + x.nsat).NonCanon(), nonmal), (y.nsat + x.sat).NonCanon(), nonmal), y.sat + x.sat);
+                }
+                case NodeType::OR_B: {
+                    auto& x = subres[0], &z = subres[1];
+                    return InputResult(z.nsat + x.nsat, Choose(Choose(z.nsat + x.sat, z.sat + x.nsat, nonmal), (z.sat + x.sat).NonCanon(), nonmal));
+                }
+                case NodeType::OR_C: {
+                    auto& x = subres[0], &z = subres[1];
+                    return InputResult(INVALID, Choose(std::move(x.sat), z.sat + x.nsat, nonmal));
+                }
+                case NodeType::OR_D: {
+                    auto& x = subres[0], &z = subres[1];
+                    auto nsat = z.nsat + x.nsat, sat_l = x.sat, sat_r = z.sat + x.nsat;
```

Comment:
In b2e3582323b1e3dfded7eb0a2597fd74ba52e4df "miniscript: satisfaction support"

This line appears to not be doing anything.

Reply:
In 3bedfb5b13e863508cb9b0d44fa03cc991151927 "miniscript: satisfaction support"

This line is still here.

Reply:
Woops. Done!